packages/playwright-core/src/server/electron/loader.ts

```diff
@@ -32,6 +32,25 @@ for (const arg of chromiumSwitches) {
   app.getAppPath = () => path.dirname(appPath);
 }
 
-(globalThis as any).__playwright_run = () => {
+let launchInfoEventPayload: any;
+app.on('ready', launchInfo => launchInfoEventPayload = launchInfo);
+
+(globalThis as any).__playwright_run = async () => {
+  // Wait for app to be ready to avoid browser initialization races.
+  await app.whenReady();
```

Comment:
This has caused a regression for apps using Electron APIs that need to be called before the `ready` event fires, such as [`protocol.registerSchemesAsPrivileged`](https://www.electronjs.org/docs/latest/api/protocol#protocolregisterschemesasprivilegedcustomschemes) and [`app.disableHardwareAcceleration`](https://www.electronjs.org/docs/latest/api/app#appdisablehardwareacceleration).

We need to give the user a chance to run synchronous code before the `ready` event fires.

- #19412